src/services/shop/shop-router.ts

```diff
@@ -42,148 +43,423 @@ shopRouter.get(
 
-        if (deleted.deletedCount == 0) {
-            return res.status(StatusCode.ClientErrorNotFound).send(ShopItemNotFoundError);
+        const userOrder = await Models.ShopOrder.findOne({ userId: userId });
+
+        // Check if user has an order
+        if (!userOrder) {
+            return res.status(StatusCode.ServerErrorInternal).send(ShopInternalError); // No order found
+        }
+
+        // Find the index of the item in the user's cart
+        const itemIndex = userOrder.items.indexOf(itemId);
+        if (itemIndex === Config.NOT_FOUND) {
+            return res.status(StatusCode.ClientErrorNotFound).send(ShopItemNotFoundError); // Item not in cart
+        }
+
+        // Update the order, decrement the quantity of the item by 1
+        const updatedShopOrder = await Models.ShopOrder.updateOne(
+            { userId: userId },
+            {
+                $inc: { [`quantity.${itemIndex}`]: -1 },
+            },
+        );
+
+        // If update fails
+        if (!updatedShopOrder) {
+            return res.status(StatusCode.ServerErrorInternal).send(ShopInternalError); // Internal error
+        }
+
+        // If the quantity of the item becomes 0, remove the item from the cart
+        if ((userOrder.quantity?.[itemIndex] ?? 0) - 1 === 0) {
+            await Models.ShopOrder.updateOne(
+                { userId: userId },
+                {
+                    $pull: {
+                        items: itemId,
+                        quantity: 0, // Remove the corresponding quantity as well
+                    },
+                },
+            );
+        }
+
+        const updatedOrder = await Models.ShopOrder.findOne({ userId });
+        if (updatedOrder) {
+            return res.status(StatusCode.SuccessOK).json(updatedOrder);
         }
+        return res.status(StatusCode.ServerErrorInternal).send(ShopInternalError);
+    },
+);
 
-        return res.status(StatusCode.SuccessOK).send({ success: true });
+shopRouter.get(
+    "/cart",
+    specification({
+        method: "get",
+        path: "/shop/cart/",
+        tag: Tag.SHOP,
+        role: Role.USER,
+        summary: "Returns content of users cart",
+        responses: {
+            [StatusCode.SuccessOK]: {
+                description: "List of items and quantity",
+                schema: ShopOrderInfoSchema,
+            },
+            [StatusCode.ServerErrorInternal]: {
+                description: "Errors that should never happen",
+                schema: ShopInternalErrorSchema,
+            },
+        },
+    }),
+    async (req, res) => {
+        const { id: userId } = getAuthenticatedUser(req);
+
+        //get their order from order db
+        let userOrder = await Models.ShopOrder.findOne({ userId: userId });
+        if (!userOrder) {
+            const shopOrder: ShopOrder = {
+                items: [],
+                quantity: [],
+                userId: userId,
+            };
+
```

Comment:
Again, whenever the order is gotten we need to verify it's still valid (checking quanity && points)